transport-native-epoll/src/main/java/io/netty/channel/epoll/AIOContext.java

```diff
@@ -109,13 +138,40 @@ public void processReady(AIOEpollFileChannel file) {
                     throw new IllegalStateException("" + id + " missing");
                 }
 
+                // Finally complete the request
                 int len = (int) lengthRead;
                 assert len == lengthRead;
                 assert len >= 0 : len;
                 request.buffer.limit(len);
                 request.buffer.position(len);
                 request.handler.completed(len, request.attachment);
             }
+
+            //add any pending reads
+            //We do this after processing the finished reads incase any outstanding reads
+            //they share the same read buffer
+            for (int i = 0; i < numReady; i++) {
+                --outstanding;
```

Comment:
Here I have a problem with the `outstanding` counter again because an exception will cause this counter not to be updated whereas the requests are no longer outstanding.